ShowMeTheRoapes/Paracord
# Dockerizing

## Proposed Changes
- Dockerized the ui and api projects
- Dockerized an instance of the mongo db so that we can have a local instance running for testing.

## Testing
- Pull down project
- Follow ReadMe instructions for starting up the project (add a -d to the up command to run it in a detatched mode)
- Make sure all of the docker containers start in an ok status by running `docker ps`. There should be 3 containers.
- Do a GET request to `localhost:8080/api/v1/strands` should return `[]` with 200 status.
- POST request to `localhost:8080/api/v1/strands` with body below should return a 201 created status.
```json
{
	"ipAddress":"127.0.0.1",
	"name": "TestStrand",
	"port": 1234
}
```
- Do the same GET request as above and you should see one strand.
- Do a `docker-compose down` to wipe away all of the containers.
- Do another `docker-compose up` to create new containers.
- Do the same GET request to see that the data persisted.
